### PR TITLE
Feature/jobs filtering re-design

### PR DESCRIPTION
**Pull request by dorilla**

### CODE REVIEW

**Comment by seuros** [comment]
Could you provide screenshots ?

**Comment by dorilla** [comment]
Here is the wireframe from Assembly 
image

Here is the design implemented image

**Comment by dorilla** [comment]
@seuros build failed after pulling master

**Comment by seuros** [comment]
Could you rebase and squash commits ?

File: config/initializers/string_extension.rb [hunk 1]
```diff
@@ -2,4 +2,19 @@
...
+  # Public: Converts a string to a boolean.
+  #   Handles various boolean synonyms as well (eg. 't' or 'true' or 'yes', etc...)
+  #
+  # Examples
+  #
+  #   "yes".to_bool
+  #   # => true
+  #
+  # Returns the converted boolean.
+  def to_bool
```
Comment (seuros) [inline config/initializers/string_extension.rb]: I'm not a fan of patching String. if we are going to raise an error.

Comment (dorilla) [inline config/initializers/string_extension.rb]: i see... how would you suggest we handle an argument like "neutral".to_bool ? default to nil?

**Comment by dorilla** [comment]
@seuros went back a commit, and tests are passing

**Comment by rohitpaulk** [comment]
@dorilla - IIRC the failure was due to a connection timeout.

File: config/initializers/string_extension.rb (hunk 1 above)
Comment (seuros) [inline config/initializers/string_extension.rb]: something simple like 
```ruby
@remote_allowed = params[:remote] == 'true'
```

Comment (dorilla) [inline config/initializers/string_extension.rb]: ok

Comment (dorilla) [inline config/initializers/string_extension.rb]: @seuros 081415f removes the method

File: app/controllers/opportunities_controller.rb [hunk 2]
```diff
@@ -76,21 +76,21 @@ def index
     chosen_location = (params[:location] || closest_to_user(current_user)).try(:titleize)
     chosen_location = nil if chosen_location == 'Worldwide'
 
+    @remote_allowed = params[:remote]     == 'true' || params[:remote]     == true
```
Comment (seuros) [inline app/controllers/opportunities_controller.rb]: i don't think we will have a casted value ever.

Comment (dorilla) [inline app/controllers/opportunities_controller.rb]:
```Started GET "/jobs?q=&location=Worldwide&fulltime=true&commit=search" for 10.0.2.2 at 2015-01-02 20:18:46 +0000
Processing by OpportunitiesController#index as HTML
  Parameters: {"q"=>"", "location"=>"Worldwide", "fulltime"=>"true", "commit"=>"search"}
```

query string params come in as strings

Comment (seuros) [inline app/controllers/opportunities_controller.rb]: yes, that what i mean. It should test with string only.

Comment (dorilla) [inline app/controllers/opportunities_controller.rb]: done

File: app/assets/stylesheets/application.css.scss [hunk 3]
```diff
@@ -2045,3 +2045,116 @@ input[type=file].safari5-upload-hack {
...
+
+.custom-select::after {
+  content: "";
+  position: absolute;
+  width: 9px;
+  height: 8px;
+  top: 50%;
+  right: 1em;
+  margin-top:-4px;
+  background-image: url(http://filamentgroup.com/files/select-arrow.png);
```
Comment (seuros) [inline app/assets/stylesheets/application.css.scss]: let have this file locally.

Comment (dorilla) [inline app/assets/stylesheets/application.css.scss]: oh yes, i did mean to do that

File: app/controllers/opportunities_controller.rb [hunk 4]
```diff
@@ -76,21 +76,21 @@ def index
     chosen_location = (params[:location] || closest_to_user(current_user)).try(:titleize)
     chosen_location = nil if chosen_location == 'Worldwide'
 
+    @remote_allowed = params[:remote]     == 'true'
+    @fulltime       = params[:fulltime]   == 'true'
```
Comment (seuros) [inline app/controllers/opportunities_controller.rb]: do we really need ivars ?

File: app/controllers/opportunities_controller.rb [hunk 5]
```diff
@@ -164,18 +164,18 @@ def geocode_location(location)
     Rails.cache.fetch("geocoded_location_of_#{location}") { User.where('LOWER(city) = ?', location.downcase).map { |u| [u.lat, u.lng] }.first || [0.0, 0.0] }
   end
 
-  def get_jobs_for(chosen_location, tag, page)
+  def get_jobs_for(chosen_location, tag, page, q = nil, remote_allowed = false, fulltime = false)
```
Comment (seuros) [inline app/controllers/opportunities_controller.rb]: what is q ?

Comment (dorilla) [inline app/controllers/opportunities_controller.rb]: the keyword query

File: app/controllers/opportunities_controller.rb (hunk 4 above)
Comment (dorilla) [inline app/controllers/opportunities_controller.rb]: it's used to initialize the checkboxes

File: app/controllers/opportunities_controller.rb (hunk 5 above)
Comment (seuros) [inline app/controllers/opportunities_controller.rb]: could we use query instead ? The code should be clear. The param can stay as it is.

File: app/controllers/opportunities_controller.rb (hunk 4 above)
Comment (seuros) [inline app/controllers/opportunities_controller.rb]: :+1:

File: app/controllers/opportunities_controller.rb (hunk 5 above)
Comment (dorilla) [inline app/controllers/opportunities_controller.rb]: sure

Comment (dorilla) [inline app/controllers/opportunities_controller.rb]: done

File: app/assets/stylesheets/application.css.scss (hunk 3 above)
Comment (dorilla) [inline app/assets/stylesheets/application.css.scss]: done

File: spec/controllers/opportunity_controlller_spec.rb [hunk 6]
```diff
@@ -13,12 +13,38 @@
       expect(response).to render_template(['opportunities/index', 'layouts/jobs'])
     end
 
-    context "when it's filtered by remote opportunities" do
-      before { @opportunity = Fabricate(:opportunity, remote: true, location: "Anywhere") }
+    context "when it's filtered" do
+      context "by remote opportunities" do
+        before(:all) { @opportunity1 = Fabricate(:opportunity, remote: true, location: "Anywhere") }
```
Comment (seuros) [inline spec/controllers/opportunity_controlller_spec.rb]: we need to add a job that won't be fetched.

**Pull request merged**